rpcs3/Emu/Cell/Modules/cellSpurs.cpp

```diff
@@ -2408,21 +2456,30 @@ s32 _spurs::add_workload(ppu_thread& ppu, vm::ptr<CellSpurs> spurs, vm::ptr<u32>
 			spurs->wklEvent2[index] |= 2;
 		}
 
-		spurs->wklIdleSpuCountOrReadyCount2[wnum] = 0;
+		spurs->wklIdleSpuCountOrReadyCount2[wnum].release(0);
 	}
 
-	vm::atomic_op(spurs->wklMaxContention[index], [&](u8& v)
+	spurs->wklMaxContention[index].atomic_op([&](u8& v)
 	{
 		v &= (wnum <= 15 ? 0xf0 : 0x0f);
 		v |= (maxContention > 8 ? 8 : maxContention) << (wnum < CELL_SPURS_MAX_WORKLOAD ? 0 : 4);
 	});
 
-	vm::atomic_op<true>((wnum <= 15 ? spurs->wklSignal1 : spurs->wklSignal2), [&](be_t<u16>& data)
+	(wnum <= 15 ? spurs->wklSignal1 : spurs->wklSignal2).atomic_op([&](be_t<u16>& data)
 	{
 		data &= ~(0x8000 >> index);
 	});
 
-	spurs->wklFlagReceiver.compare_and_swap(wnum, 0xff);
+	if (spurs->wklFlagReceiver == wnum)
```

Comment:
Why if..cmpxchg?

Reply:
Relaxed operation if succeeds, better performance.

Reply:
I get that, but I'm wondering if the optimization is worth it. Mostly for 2 reasons, I don't know often this call is made say in one second, and also is regular cmpxchg that slow? I know cmpxchg16b is bad, but this is the regular 4 byte version. Is it that much more likely that the condition is false? I'm not familiar with the HLE cellSpurs methods.

Reply:
Maybe a comment there would help.